src/test/scala/chiselTests/experimental/hierarchy/DefinitionSpec.scala

```diff
@@ -0,0 +1,431 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+package experimental.hierarchy
+
+import chisel3._
+import chisel3.experimental.BaseModule
+import chisel3.experimental.hierarchy.{Definition, Instance, instantiable, public}
+
+// TODO/Notes
+// - In backport, clock/reset are not automatically assigned. I think this is fixed in 3.5
+// - CircuitTarget for annotations on the definition are wrong - needs to be fixed.
+class DefinitionSpec extends ChiselFunSpec with Utils {
+  import Annotations._
+  import Examples._
+  describe("0: Definition instantiation") {
+    it("0.0: module name of a definition should be correct") {
+      class Top extends Module {
+        val definition = Definition(new AddOne)
+      }
+      check(new Top(), Nil, Seq("  module AddOne :"))
```

Comment:
Nitty (and clarified later), but it's confusing to me what this is "checking". I'd prefer for these tests to be written as:

```scala
val (circuit, annotations) = compile(foo)
circuit should contain (bar)
annotations should contain (baz)
```

I'd prefer for tests to be nicely encapsulated.

Reply:
I agree with all of this, I'll update them to more of this style.